aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -702,7 +730,9 @@ void TensorIterator::check_mem_overlaps() {
     assert_no_internal_overlap(output);
     for (int j = num_outputs_; j < ntensors(); j++) {
       const auto& input = operands_[j].tensor;
-      assert_no_partial_overlap(output, input);
+      if (!output.is_same(input)) {
```

Comment:
Should this check be sunk into `assert_no_partial_overlap`?

Reply:
I already added a short-circuit in get_overlap_status, but it was behind .contiguous so was slower, now I pulled it up there and removed from here, seems to be ok.